gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -404,10 +409,51 @@ public void delete(URI path, boolean recursive) throws IOException {
               : listFileInfoForPrefixPage(
                       fileInfo.getPath(), DELETE_RENAME_LIST_OPTIONS, /* pageToken= */ null)
                   .getItems();
-      if (!itemsToDelete.isEmpty() && !recursive) {
+
+      /*TODO : making listing of folder and object resources in parallel*/
+      if (isHnBucket) {
+
+        /**
+         * Get list of folders in path if the bucket is HN enabled bucket For recursive delete, get
+         * all folder resources. For non-recursive delete, only get maximum of 1 folder resources
+         * other than the path.
+         */
+        listOfFolders =
+            recursive
+                ? listFoldersInfoForPrefixPage(
+                        fileInfo.getPath(),
+                        ListFolderOptions.DEFAULT_FOLDER_LIST,
+                        /* pageToken= */ null)
+                    .getItems()
+                : Arrays.asList(
```

Comment:
If the URI is a bucket, will it delete the bucket?

Reply:
No, deleteFolder will not be called for a bucket. There are checks for it.